### PR TITLE
Japanese Hepburn and Kunrei-shiki MEXT maps

### PR DESCRIPTION
**Pull request by ronaldtse**

### CODE REVIEW

**Comment by ronaldtse** [comment]
Rebased but still issues:

3 is due to the first character is a Unicode character made up of two characters. The replacement of "が" is "ga", but "か" is "ka". It treated the extra dots to the "a゙". 

The problem is the character replacement should work on Unicode character boundaries, not fixed-width characters.

```
  3) Interscript maps/mext-jpn-Hrkt-Latn-hepburn.yaml system test for {"source"=>"がっこう", "expected"=>"gakkō"}
     Failure/Error: expect(result).to eq(test["expected"])
     
       expected: "gakkō"
            got: "ka゙kkō"
     
       (compared using ==)
     # ./spec/interscript_spec.rb:13:in `block (5 levels) in <top (required)>'
```

4 seems to be the failure to run the post-processing more than once.

```
  4) Interscript maps/mext-jpn-Hrkt-Latn-hepburn.yaml system test for {"source"=>"ぽっぽっや", "expected"=>"poppoyya"}
     Failure/Error: expect(result).to eq(test["expected"])
     
       expected: "poppoyya"
            got: "poppoっya"
     
       (compared using ==)
     # ./spec/interscript_spec.rb:13:in `block (5 levels) in <top (required)>'
```

File: maps/mext-jpn-Hrkt-Latn-hepburn.yaml [hunk 1]
```diff
@@ -48,24 +48,57 @@ tests:
...
+    - pattern: ッ([BbCcDdFfGgHhJjKkLlMmNnPpQqRrSsTtVvXxZz]) # ッ folloved by consonant
+      result: "\\1\\1"
+
+    # Note 2. はねる音を表わすnと次にくる母音字またはyとを切り離す必要がある場合には、nの次に’を入れる。
+    # - pattern: "([aeiou])n"
+    #   result: "\\1'n"
+    - pattern: "n([aeiouāīūēō])"
+      result: "n'\\1"
+    - pattern: "ny"
+      result: "n'y"
```
Comment (ronaldtse) [inline maps/mext-jpn-Hrkt-Latn-hepburn.yaml:80]: This rule presents some problem for us:

```ruby
 7) Interscript maps/mext-jpn-Hrkt-Latn-hepburn.yaml system test for {"source"=>"ぽっぽっや", "expected"=>"poppoyya"}
 Failure/Error: expect(result).to eq(test["expected"])

   expected: "poppoyya"
        got: "poppoっya"

   (compared using ==)
 # ./spec/interscript_spec.rb:13:in `block (5 levels) in <top (required)>'

 8) Interscript maps/mext-jpn-Hrkt-Latn-hepburn.yaml system test for {"source"=>"にゃっほー", "expected"=>"nyahhō"}
 Failure/Error: expect(result).to eq(test["expected"])

   expected: "nyahhō"
        got: "n'yahhō"

   (compared using ==)
 # ./spec/interscript_spec.rb:13:in `block (5 levels) in <top (required)>'
```

If the characters ("kin yu" => "kin'yu") are separate, this rule works.

If the characters are not originally separate, e.g. "にゃっ" ("nya"), this rule should not apply.

Maybe we should have some language that describes the transliteration character boundaries? e.g. "にゃっ" is one single transliteration unit. "き" is also one single transliteration unit.

**Comment by ronaldtse** [comment]
@chaaklau do we still need these or are they already implemented in the other PRs? Thanks!

**Comment by chaaklau** [comment]
> @chaaklau do we still need these or are they already implemented in the other PRs? Thanks!

This has been done in #210, closing this.

**Pull request closed without merging**